### PR TITLE
[WIP] gh-106608: Have `_PyUOpExecutorObject` use a variable-length array

### PR DESCRIPTION
**Pull request by justdan6**
Attempting to work out gh-106608

### CODE REVIEW

File: Python/executor.c [hunk 1]
```diff
@@ -31,6 +31,39 @@
 #define ENABLE_SPECIALIZATION 0
 
 
+// TODO: continue editing this from tupleobject.c to fit _PyUOpExecutorObject
+static _PyUOpExecutorObject *
+tuple_alloc(Py_ssize_t size)
+{
+    _PyUOpExecutorObject *op = maybe_freelist_pop(size);
```
Comment (Fidget-Spinner) [inline Python/executor.c]: freelists are an optimization. You could remove this line for now (and the branch that checks for `NULL` below

File: Python/executor.c [hunk 2]
```diff
@@ -31,6 +31,39 @@
...
+    if (op == NULL) {
+        /* Check for overflow */
+        if ((size_t)size > ((size_t)PY_SSIZE_T_MAX - (sizeof(PyTupleObject) -
+                    sizeof(PyObject *))) / sizeof(PyObject *)) {
+            return (_PyUOpExecutorObject *)PyErr_NoMemory();
+        }
```
Comment (Fidget-Spinner) [inline Python/executor.c]: You can remove these lines if you remove the freelist.

File: Python/executor.c [hunk 3]
```diff
@@ -31,6 +31,39 @@
...
+        op = PyObject_GC_NewVar(_PyUOpExecutorObject, &PyTuple_Type, size);
+        if (op == NULL)
+            return NULL;
```
Comment (Fidget-Spinner) [inline Python/executor.c]: These are the lines you want to keep. However, according to PEP 7, the if branch should have curly braces.

**Review by Fidget-Spinner — COMMENTED** [review]
Some early comments to help you pass CI!

File: Python/executor.c (hunk 3 above)
Comment (justdan6) [reply Python/executor.c:47]: Thank you very much for the help! 

I was looking into replacing PyTuple_Type and wasn't sure if it'd be necessary to recreate a PyTypeObject specific to this(and further potentially creating a new object) or if there's something better to use?

File: Python/executor.c (hunk 1 above)
Comment (justdan6) [reply Python/executor.c]: Thank you for the help!!

**Comment by gvanrossum** [comment]
Sorry, you can close this now -- the issue was fixed by someone else.

**Pull request closed without merging**